Cargo.toml

```diff
@@ -15,16 +15,22 @@ resolver = "2"
 [workspace.dependencies]
 
 # Async and futures crates
-futures = { version = "0.3" }
+futures = { version = "0.3", default-features = false }
```

Comment:
Wouldnt this disable features that enables channels, use of sink/stream, and other utils from its underlining crates behind its default feature? If so then we should probably enable those features thats being used here.

Reply:
You're right. I think something else enables those features so the build is not broken. That's not something we should rely on though. It seems `std` feature enables all the stuff we need.